src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1456,6 +1457,12 @@ def yaml_representer(dumper: 'yaml.SafeDumper', data: 'DaemonDescription') -> An
         return dumper.represent_dict(data.to_json().items())
 
 
+class DaemonDescriptionStatus(enum.IntEnum):
+    error = -1
+    stopped = 0
+    running = 1
```

Comment:
https://github.com/ceph/ceph/blob/05ef9e1ad99f5d8d8bf75a02103da5d98d7e1da2/qa/tasks/cephadm_cases/test_cli.py#L36

```yaml
daemon_type: osd
daemon_id: '2'
hostname: smithi142
container_id: a93bbc870330
container_image_id: 2f759791c60922712d18300934155125b3d685a60a411248cbe486ce4489bd7a
container_image_name: quay.ceph.io/ceph-ci/ceph:52fc503cf18cf3bb446b840ba00be073017b8373
container_image_digests:
- quay.ceph.io/ceph-ci/ceph@sha256:42659bfb76ef5f335b56e9a900612c30b98b66ea577069fbc2d31c94a9b434c6
version: 17.0.0-699-g52fc503c
status: !!python/object/apply:orchestrator._interface.DaemonDescriptionStatus
- 1
status_desc: running
osdspec_affinity: None
is_active: false
last_refresh: '2021-02-12T16:38:43.947156Z'
created: '2021-02-12T16:30:13.999057Z'
started: '2021-02-12T16:30:13.739594Z'
```
```
sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph log 'Ended test tasks.cephadm_cases.test_cli.TestCephadmCLI.test_yaml'
test_yaml (tasks.cephadm_cases.test_cli.TestCephadmCLI) ... FAIL

======================================================================
FAIL: test_yaml (tasks.cephadm_cases.test_cli.TestCephadmCLI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_ceph-c_52fc503cf18cf3bb446b840ba00be073017b8373/qa/tasks/cephadm_cases/test_cli.py", line 36, in test_yaml
    self.assertNotIn('!!python', out)
``` 

https://pulpito.ceph.com/swagner-2021-02-11_11:00:52-rados:cephadm-wip-swagner3-testing-2021-02-10-1322-distro-basic-smithi/5874628

I **never** thought this dead simple test would save us 8 months later

Reply:
i removed the DaemonDescriptionStatus changes as i could not get this to work and would like to get this merged and that was not really related to the main goal of this pr. changes are here for future reference https://github.com/Daniel-Pivonka/ceph/commit/bd92843d0a59cf2e90227e1e39b5cabef7dcdbf3